docs/architecture/adr-104-out-of-band-state-sync.md

```diff
@@ -0,0 +1,215 @@
+# ADR 104: Out-of-band state sync support
+
+## Changelog
+
+- 2023-05-05: Initial Draft (@cason)
+
+## Status
+
+Accepted | Rejected | Deprecated | Superseded by
+
+## Context
+
+> 1. What is state sync?
+
+From CometBFT v0.34, synchronizing a fresh node with the rest of the network
+can benefit from [state sync][state-sync], a protocol for discovering,
+fetching, and installing application snapshots.
+State sync is able to rapidly bootstrap a node by installing a relatively
+recent state machine snapshot, instead of replaying all historical blocks.
+
+> 2. What is the problem?
+
+With the widespread adoption of State sync to bootstrap nodes, however,
+what should be one of its strengths - the ability to discover and fetch
+application snapshots from peers in the p2p network - has turned out to be one
+of its weakness.
+In fact, while downloading recent snapshots is very convenient for new nodes
+(clients of the protocol), providing snapshots to multiple peers (as servers of
+the protocol) is _bandwidth-consuming_.
```

Comment:
i would say this is less of a bandwidth issue, but more of a reason of space and system requirements.